xpk.py

```diff
@@ -2124,7 +2158,7 @@ def get_all_subnets_programmatic(args) -> tuple[list[str], int]:
     List of subnets and 0 if successful and 1 otherwise.
   """
   command = (
-      'gcloud compute networks subnets list'
+      f'gcloud compute networks subnets list --filter=region:{zone_to_region(args.zone)}'
```

Comment:
I think you also want `--project={args.project}` here as well so that xpk project arg is propagated to this command as well.